src/interface.jl

```diff
@@ -245,7 +271,15 @@ julia> has_vertex(SimpleGraph(2), 3)
 false
 ```
 """
-has_vertex(x, v) = _NI("has_vertex")
+function has_vertex end
+
+@traitfn function has_vertex(g::G::(!HasContiguousVertices), v) where {G}
+    return _NI("has_vertex")
+end
+
+@traitfn function has_vertex(g::G::(HasContiguousVertices), v) where {G}
+    return 0 < v <= nv(g)
```

Comment:
can we do `return v ϵ vertices(g)` instead? There may be optimizations for UnitRanges here.